packages/flutter/lib/src/material/icon_button.dart

```diff
@@ -719,23 +719,26 @@ class IconButton extends StatelessWidget {
         effectiveIcon = selectedIcon!;
       }
 
-      Widget iconButton = effectiveIcon;
-      if (tooltip != null) {
-        iconButton = Tooltip(
-          message: tooltip,
-          child: effectiveIcon,
-        );
-      }
-
-      return _SelectableIconButton(
+      Widget iconButton = _SelectableIconButton(
         style: adjustedStyle,
         onPressed: onPressed,
         autofocus: autofocus,
         focusNode: focusNode,
         isSelected: isSelected,
         variant: _variant,
-        child: iconButton,
+        child: effectiveIcon,
       );
+
+      if (tooltip != null) {
+        iconButton = MergeSemantics(
```

Comment:
Just curious, why do we need to merge semantics here?

Reply:
I see! It seems because `ButtonStyleButton` creates a separate semantic node by using `container: true`, so when we use `Tooltip` to wrap the M3 `IconButton` which is a subclass of `ButtonStyleButton`, tooltip becomes another node instead of part of the semantic node of `IconButton`. This implementation is good for now because it seems other common buttons don't need tooltips now. If we have more tooltip requests for other ButtonStyleButtons, we can introduce tooltip property in `ButtonStyleButton` and pass it to all other subclasses, including `IconButton`:)

Reply:
I agree! Fingers crossed MergedSemantics doesn't cause any regressions.